src/main/java/net/openhft/chronicle/core/util/StringUtils.java

```diff
@@ -352,6 +360,8 @@ private static void ensureJava9Plus() {
     @Java9
     @NotNull
     public static String newStringFromBytes(byte @NotNull [] bytes) {
+        if (!HAS_ONE_BYTE_PER_CHAR)
```

Comment:
I appreciate that branch prediction is incredibly effective, but would it not be more JIT-friendly if instead of this conditional we created a static final `Function<String, byte[]>' in the static block and called that?